### ISSUE TITLE
Why didn't torch.no_grad() and model.eval() take effects in inference Yolov5?

### ISSUE THREAD

**Issue by melonux** [ultralytics/yolov5#649]
## ❔Question

I trained a small model (yolov5s.yaml), and tried to inference objects in videos (800x480) by device=cpu.  It took 0.2 seconds for each frame, and use about 650MB memory.
After using model.eval() and torch.no_grad(), there is *nothing* gain in speed or memory. But why?

I also implement batch detect just like your test.py:

        batch_img = np.stack(frames, axis=0)

        with torch.no_grad():
            pred = model(batch_img, augment=False)[0]
            pred_result = non_max_suppression(pred, conf_thres, iou_thres)

Memory usage grew up linearly with the number of frames in the batch. Is there any way to decrease memory usage in inference?

Thank you in advance.

**Comment by glenn-jocher** [comment]
I don't see anything out of the ordinary in your comments. They also do not pertain to YOLOv5 specifically, you may want to ask them on the pytorch forums instead.
